### PR TITLE
Fea/use authentication

### PR DESCRIPTION
**Pull request by shaybs247**

### CODE REVIEW

File: src/utils/useAuthentication/index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
```
Comment (deaBrgil) [inline src/utils/useAuthentication/index.ts:1]: Can you move it to the hooks folder? 😁

File: src/utils/useAuthentication/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import { logout, startTTH, TStartTTH } from '../TwoTokenHandshake';
+
+export const useAuthentication = (authentication: Partial<TStartTTH>) => {
+  const { userId, performBEDHandshakeCallback } = authentication;
```
Comment (deaBrgil) [inline src/utils/useAuthentication/index.ts:6]: Can we maybe change `performBEDHandshakeCallback` to something more declarative? Like `onUserChange` or `onAuthChange`? Not super important though...

Comment (deaBrgil) [inline src/utils/useAuthentication/index.ts:6]: Can we maybe change `performBEDHandshakeCallback` to something more declarative? Like `onUserChange` or `onAuthChange`? Not super important though...

**Review by deaBrgil — APPROVED** [review]

File: src/utils/useAuthentication/index.ts (hunk 1 above)
Comment (shaybs247) [reply src/utils/useAuthentication/index.ts:1]: sure

**Review by deaBrgil — APPROVED** [review]

**Pull request merged**